analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -112,8 +123,37 @@ public Verifier(VerifierBuilder builder)
             }
         }
 
-        private IEnumerable<Compilation> Compile() =>
-            CreateProject(builder.ConcurrentAnalysis).Solution.Compile(builder.ParseOptions.ToArray());
+        public void VerifyUtilityAnalyzerProducesEmptyProtobuf()     // This should never has any arguments
+        {
+            foreach (var compilation in Compile(false))
+            {
+                DiagnosticVerifier.Verify(compilation, analyzers.Single(), CompilationErrorBehavior.Default);
+                new FileInfo(builder.ProtobufPath).Length.Should().Be(0, "protobuf file should be empty");
+            }
+        }
+
+        public void VerifyUtilityAnalyzer<TMessage>(Action<IReadOnlyList<TMessage>> verifyProtobuf)
```

Comment:
This seems an over-complicated way to check the protobuf file contents. I suggest replacing this with a method that returns the content and lets the tests make the checks directly.

Reply:
I think that in the consumers we will replace code like:
```
            new VerifierBuilder().AddAnalyzer(() => new DummyUtilityAnalyzerCS(protobufPath, message)).AddSnippet("// Nothing to see here").WithProtobufPath(protobufPath)
                .VerifyUtilityAnalyzer<LogInfo>(x =>
                {
                    x.Should().ContainSingle().Which.Text.Should().Be("Lorem Ipsum");
                    wasInvoked = true;
                });
            wasInvoked.Should().BeTrue();
```
with
```
            new VerifierBuilder().AddAnalyzer(() => new DummyUtilityAnalyzerCS(protobufPath, message)).AddSnippet("// Nothing to see here").WithProtobufPath(protobufPath)
                .VerifyUtilityAnalyzer<LogInfo>()
                .Should().ContainSingle().Which.Text.Should().Be("Lorem Ipsum");
```

Reply:
Names can be retouched a bit too. 

I think the end result would be much easier to read and maintain.

Reply:
Discussed offline